### PR TITLE
llama70b one card to infer device map with max memory limitation 

### PR DESCRIPTION
**Pull request by Yantom1**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/text-generation/utils.py [hunk 1]
```diff
@@ -158,10 +158,19 @@ def get_torch_compiled_model(model):
...
+        with init_empty_weights():
+            model = AutoModelForCausalLM.from_config(config)
+        max_memory = {"cpu": "10GiB"}
+        device_map = infer_auto_device_map(model, max_memory=max_memory, dtype=model_dtype)
+        model = AutoModelForCausalLM.from_pretrained(args.model_name_or_path, device_map=device_map, offload_folder="/tmp/offload_folder/", offload_state_dict=True, torch_dtype=model_dtype, **model_kwargs)  
+    else: 
+        if args.peft_model is not None:
+            model = peft_model(args, model_dtype, logger, **model_kwargs)
+        else:
+            model = AutoModelForCausalLM.from_pretrained(args.model_name_or_path, torch_dtype=model_dtype, **model_kwargs)
```
Comment (regisss) [inline examples/text-generation/utils.py:173]: This means we cannot use disk offloading with PEFT models, is that intended?

**Review by regisss — COMMENTED** [review]
LGTM!
Can you also share a command that I can execute to test it please?
You will also need to run
```
pip install -U ruff
make style
```
from the root of the repo to make the code style check pass.

**Comment by Yantom1** [comment]
> LGTM! Can you also share a command that I can execute to test it please? You will also need to run
> 
> ```
> pip install -U ruff
> make style
> ```
> 
> from the root of the repo to make the code style check pass.

@regisss fixed the formation. The disk_offload is tested nightly

**Review by regisss — APPROVED** [review]
LGTM!

It would probably work the same for PEFT models where we call `merge_and_unload`.

**Pull request merged**